samples/core/blur/blur.cpp

```diff
@@ -0,0 +1,745 @@
+/*
+ * Copyright (c) 2023 The Khronos Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "blur.hpp"
+
+#include <CL/Utils/Utils.hpp>
+#include <CL/SDK/Context.hpp>
+#include <CL/SDK/CLI.hpp>
+
+// STL includes
+#include <iostream>
+#include <algorithm>
+#include <map>
+#include <math.h>
+
+// TCLAP includes
+#include <tclap/CmdLine.h>
+
+// Default image
+#include "default_image.h"
+
+std::unique_ptr<TCLAP::ValuesConstraint<std::string>> valid_op_constraint;
+
+// Add option to CLI parsing SDK utility
+template <> auto cl::sdk::parse<BlurCppExample::BlurOptions>()
+{
+    return std::make_tuple(
+        std::make_shared<TCLAP::ValueArg<std::string>>(
+            "i", "in", "Input image file", false, "", "name"),
+        std::make_shared<TCLAP::ValueArg<std::string>>(
+            "o", "out", "Output image file", false, "out.png", "name"),
+        std::make_shared<TCLAP::ValueArg<float>>("s", "size",
+                                                 "Size of blur kernel", false,
+                                                 (float)1.0, "positive float"),
+        std::make_shared<TCLAP::MultiArg<std::string>>(
+            "b", "blur", "Operation of blur to perform: box or gauss", false,
+            "box"));
+}
+
+template <>
+BlurCppExample::BlurOptions cl::sdk::comprehend<BlurCppExample::BlurOptions>(
+    std::shared_ptr<TCLAP::ValueArg<std::string>> in_arg,
+    std::shared_ptr<TCLAP::ValueArg<std::string>> out_arg,
+    std::shared_ptr<TCLAP::ValueArg<float>> size_arg,
+    std::shared_ptr<TCLAP::MultiArg<std::string>> op_arg)
+{
+    return BlurCppExample::BlurOptions{ in_arg->getValue(), out_arg->getValue(),
+                                        size_arg->getValue(),
+                                        op_arg->getValue() };
+}
+
+void BlurCppExample::single_pass_box_blur()
+{
+    std::cout << "Single-pass blur" << std::endl;
+    step++;
+
+    auto size = static_cast<cl_int>(blur_opts.size);
+    auto blur =
+        cl::KernelFunctor<cl::Memory, cl::Memory, cl_int>(program, "blur_box");
+    // blur
+    auto start = std::chrono::high_resolution_clock::now();
+    std::vector<cl::Event> passes;
+
+    auto event = blur(cl::EnqueueArgs{ queue, cl::NDRange{ width, height } },
```

Comment:
I don't see anywhere we assign to this `passes` vector.  Instead, we assign to a specific `event`.  Is this intended?  I think it will mess up the printed timings.

Reply:
This issue is still present.